examples/gno.land/p/demo/grc/grc721/nft_metadata_test.gno

```diff
@@ -0,0 +1,130 @@
+
+	dummy.mint(addr1, TokenID("1"))
+
+	_, derr := dummy.SetExtension(TokenID("1"), Metadata{
+		Name : name,
+		Description: description,
+		Image: image,
+		ImageData: imageData,
+		ExternalURL: externalURL,
+		Attributes: attributes,
+		BackgroundColor: backgroundColor,
+		AnimationURL: animationURL,
+		YoutubeURL: youtubeURL,
+	})
+
+	if derr != nil {
+		t.Errorf("Should not result in error ", derr.Error())
+	}
+
+	// Test case: Invalid token ID
+	_, err := dummy.SetExtension(TokenID("3"), Metadata{
+		Name : name,
+		Description: description,
+		Image: image,
+		ImageData: imageData,
+		ExternalURL: externalURL,
+		Attributes: attributes,
+		BackgroundColor: backgroundColor,
+		AnimationURL: animationURL,
+		YoutubeURL: youtubeURL,
+	})
+	if err != ErrInvalidTokenId {
+		t.Errorf("Expected error %v, got %v", ErrInvalidTokenId, err)
+	}
+
+	std.TestSetOrigCaller(std.Address(addr2)) // addr2
+
+	_, cerr := dummy.SetExtension(TokenID("1"), Metadata{
+		Name : name,
+		Description: description,
+		Image: image,
+		ImageData: imageData,
+		ExternalURL: externalURL,
+		Attributes: attributes,
+		BackgroundColor: backgroundColor,
+		AnimationURL: animationURL,
+		YoutubeURL: youtubeURL,
+	}) // addr2 trying to set Extension for token 1
+	if cerr != ErrCallerIsNotOwner {
+		t.Errorf("Expected error %v, got %v", ErrCallerIsNotOwner, cerr)
+	}
+
+	// Test case: Retrieving Extension
+	std.TestSetOrigCaller(std.Address(addr1)) // addr1
+
+	dummyExtension, err := dummy.Extension(TokenID("1"))
+	if err != nil {
+		t.Errorf("Extension error: %v, ", err.Error())
+
+	} else {
+		if dummyExtension.Image != image {
+			t.Errorf("Expected Extension's image %v, got %v", image, dummyExtension.Image)
+		}
+		
+		if dummyExtension.ImageData != imageData {
+			t.Errorf("Expected Extension's imageData %v, got %v", imageData, dummyExtension.ImageData)
+		}
+		
+		if dummyExtension.ExternalURL != externalURL {
+			t.Errorf("Expected Extension's externalURL %v, got %v", externalURL, dummyExtension.ExternalURL)
+		}
+		
+		if dummyExtension.Description != description {
+			t.Errorf("Expected Extension's description %v, got %v", description, dummyExtension.Description)
+		}
+	
+		if dummyExtension.Name != name {
+			t.Errorf("Expected Extension's name %v, got %v", name, dummyExtension.Name)
+		}
+		
+		if len(dummyExtension.Attributes) != len(attributes) {
+			t.Errorf("Expected %d Extension's attributes %v, got %v", len(attributes), len(dummyExtension.Attributes))
+		}
+		 
+		if dummyExtension.BackgroundColor != backgroundColor {
+			t.Errorf("Expected Extension's backgroundColor %v, got %v", backgroundColor, dummyExtension.BackgroundColor)
+		}
+		
+		if dummyExtension.AnimationURL != animationURL {
+			t.Errorf("Expected Extension's animationURL %v, got %v", animationURL, dummyExtension.AnimationURL)
+		}
+		
+		if dummyExtension.YoutubeURL != youtubeURL {
+			t.Errorf("Expected Extension's youtubeURL %v, got %v", youtubeURL, dummyExtension.YoutubeURL)
+		}
```

Comment:
Currently, the `ufmt.Errorf` function is implemented throught the `ufmt.Sprintf` function. But this function does not support the `%v` formatting verbs for now. Please refer to the link below for the list of available verbs.

https://github.com/gnolang/gno/blob/2f72f843d26d8fc3deb694c4e4812f22c8096866/examples/gno.land/p/demo/ufmt/ufmt.gno#L45-L53

Reply:
i got it